src/main/java/net/snowflake/client/core/arrow/tostringhelpers/ArrowArrayStringRepresentationBuilder.java

```diff
@@ -0,0 +1,27 @@
+package net.snowflake.client.core.arrow.tostringhelpers;
+
+import net.snowflake.client.core.SnowflakeJdbcInternalApi;
+import net.snowflake.client.jdbc.SnowflakeType;
+
+@SnowflakeJdbcInternalApi
+public class ArrowArrayStringRepresentationBuilder extends ArrowStringRepresentationBuilderBase {
+
+  public ArrowArrayStringRepresentationBuilder() {
+    super();
+    this.append('[');
+  }
+
+  public ArrowStringRepresentationBuilderBase appendValue(String value, SnowflakeType valueType) {
+    super.addCommaIfNeeded();
+    if (shouldQuoteValue(valueType)) {
```

Comment:
I think it might be worth moving this logic to the base class and adding .append(value, type). We repeat this check in both subclasses, and it could happen more in the future.

Reply:
good catch, will move